File: 0001-0100/0001/twoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}                                    
        for i, num in enumerate(nums):              
            targ_num = target - num                 
            if targ_num in dic:                     
                return [dic[targ_num], i]           
            dic[num] = i                            

# Test
nums = [2, 7, 11, 15]
target = 9

solution = Solution()

print(solution.twoSum(nums, target))

File: 0001-0100/0001/twoSum.java
import java.util.HashMap;

class twoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int targNum = target - num;
            if (map.containsKey(targNum)) {
                return new int[]{map.get(targNum), i};
            }
            map.put(num, i);
        }

        return new int[]{-1, -1};   //If no solution is found
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        twoSum solution = new twoSum();
        int[] result = solution.twoSum(nums, target);
        System.out.println("[" + result[0] + ", " + result[1] + "]");
    }
}
